### PR TITLE
Allow to initialize web3wallet and web3Modal sdks

### PR DESCRIPTION
**Pull request by kacperoak**

### CODE REVIEW

File: product/walletconnectmodal/src/main/kotlin/com/walletconnect/wcmodal/client/WalletConnectModal.kt [hunk 1]
```diff
@@ -38,62 +42,49 @@ object WalletConnectModal {
...
+        onSuccess: () -> Unit = {},
+        onError: (Modal.Model.Error) -> Unit
+    ) {
+        this.excludedWalletsIds = init.excludedWalletIds
+        this.recommendedWalletsIds = init.recommendedWalletsIds
+        _sessionParams = init.sessionParams
+        runCatching {
+            wcKoinApp.modules(walletConnectModalModule())
+            setDelegate(WalletConnectModalDelegate)
+        }.onFailure { error -> onError(Modal.Model.Error(error)) }
```
Comment (TalhaAli00) [inline product/walletconnectmodal/src/main/kotlin/com/walletconnect/wcmodal/client/WalletConnectModal.kt:68]:
```suggestion
        }.onFailure { error -> return@onInitializedClient onError(Modal.Model.Error(error)) }
```

File: product/web3modal/src/main/kotlin/com/walletconnect/web3/modal/client/Web3Modal.kt [hunk 2]
```diff
@@ -61,21 +64,33 @@ object Web3Modal {
...
+        init: Modal.Params.Init,
+        onSuccess: () -> Unit = {},
+        onError: (Modal.Model.Error) -> Unit
+    ) {
+        this.excludedWalletsIds = init.excludedWalletIds
+        this.recommendedWalletsIds = init.recommendedWalletsIds
+        runCatching {
+            wcKoinApp.modules(web3ModalModule())
+            setDelegate(Web3ModalDelegate)
+        }.onFailure { error -> onError(Modal.Model.Error(error)) }
```
Comment (TalhaAli00) [inline product/web3modal/src/main/kotlin/com/walletconnect/web3/modal/client/Web3Modal.kt:90]:
```suggestion
        }.onFailure { error -> return@onInitializedClient onError(Modal.Model.Error(error)) }
```

**Review by TalhaAli00 — APPROVED** [review]

File: product/web3modal/src/main/kotlin/com/walletconnect/web3/modal/client/Web3Modal.kt [hunk 3]
```diff
@@ -61,21 +64,33 @@ object Web3Modal {
...
+                onInitializedClient(init, onSuccess, onError)
             },
             onError = { error ->
-                onError(Modal.Model.Error(error.throwable))
-                return@initialize
+                if (error.throwable is SignClientAlreadyInitializedException) {
+                    onInitializedClient(init, onSuccess, onError)
+                } else {
+                    onError(Modal.Model.Error(error.throwable))
+                    return@initialize
```
Comment (Elyniss) [inline product/web3modal/src/main/kotlin/com/walletconnect/web3/modal/client/Web3Modal.kt:74]:
```suggestion
                   return@initialize  onError(Modal.Model.Error(error.throwable))
```

**Review by Elyniss — APPROVED** [review]
LGTM only some small nitpicks

**Pull request merged**